driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -164,14 +167,60 @@ public RequestHandler(
     ConsistencyLevel consistency = statement.getConsistencyLevel();
     if (consistency == null) consistency = Statement.DEFAULT.getConsistencyLevel();
 
-    String statementType = "regular";
-    if (statement instanceof BatchStatement) statementType = "batch";
-    else if (statement instanceof PreparedStatement) statementType = "prepared";
+    String statementType = null;
+    String statementText = null;
+    int batchSize = 1;
+
+    String keyspace = null;
+    String partitionKey = null;
+    String table = null;
+
+    if (statement instanceof BatchStatement) {
+      statementType = "batch";
+      batchSize = ((BatchStatement) statement).size();
+      StringBuilder statementTextBuilder = new StringBuilder(STATEMENT_MAX_LENGTH);
+      for (Statement subStatement : ((BatchStatement) statement).getStatements()) {
+        if (subStatement instanceof BoundStatement)
+          statementTextBuilder.append(((BoundStatement) subStatement).statement.getQueryString());
+        else statementTextBuilder.append(subStatement.toString());
+      }
+      statementText = statementTextBuilder.toString();
+    } else if (statement instanceof BoundStatement) {
+      statementType = "prepared";
+      statementText = ((BoundStatement) statement).statement.getQueryString();
+      keyspace = ((BoundStatement) statement).getKeyspace();
+
+      ColumnDefinitions boundColumns =
+          ((BoundStatement) statement).statement.getPreparedId().boundValuesMetadata.variables;
+
+      StringBuilder partitionKeyBuilder = new StringBuilder(PARTITION_KEY_MAX_LENGTH);
+      for (int i = 0; i < boundColumns.size(); i++) {
+        String columnName = boundColumns.getName(i);
+        Object value = ((BoundStatement) statement).getObject(i);
+        String valueString = value.toString();
+        partitionKeyBuilder.append(columnName);
+        partitionKeyBuilder.append('=');
+        partitionKeyBuilder.append(valueString);
+        partitionKeyBuilder.append(',');
+      }
+      partitionKey = partitionKeyBuilder.toString();
+      if (boundColumns.size() > 0) table = boundColumns.getTable(0);
+    } else if (statement instanceof RegularStatement) {
+      statementType = "regular";
+      statementText = ((RegularStatement) statement).toString();
+    }
 
     this.tracingInfo = tracingInfo;
     this.tracingInfo.setNameAndStartTime("request");
     this.tracingInfo.setConsistencyLevel(consistency);
-    this.tracingInfo.setStatementType(statementType);
+    this.tracingInfo.setRetryPolicy(retryPolicy());
+    this.tracingInfo.setBatchSize(batchSize);
```

Comment:
Shouldn't it be set for BatchStatements only?

Reply:
Only keyspace, partitionKey ans table name as Piotrek said